source/include/ice_api.h

```diff
@@ -0,0 +1,105 @@
+                                    const uint32_t priority );
+
+IceResult_t Ice_CreateCandidatePair( IceAgent_t * pIceAgent,
+                                     IceCandidate_t * pLocalCandidate,
+                                     IceCandidate_t * pRemoteCandidate );
+
+IceResult_t Ice_UpdateSrflxCandidateAddress( IceAgent_t * pIceAgent,
+                                             IceCandidate_t * pCandidate,
+                                             const IceIPAddress_t * pIpAddr );
+
+IceResult_t Ice_InitializeStunPacket( IceAgent_t * pIceAgent,
+                                      StunContext_t * pStunCxt,
+                                      uint8_t * pTransactionId,
+                                      uint8_t * pStunMessageBuffer,
+                                      StunHeader_t * pStunHeader,
+                                      uint8_t isGenerateTransactionID,
+                                      uint8_t isStunBindingRequest );
+
+IceResult_t Ice_PackageStunPacket( IceAgent_t * pIceAgent,
+                                   StunContext_t * pStunCxt,
+                                   uint8_t * pPpassword,
+                                   uint32_t passwordLen,
+                                   uint32_t * pStunMessageBufferLength );
+
+IceResult_t Ice_CreateRequestForSrflxCandidate( IceAgent_t * pIceAgent,
+                                                uint8_t * pStunMessageBuffer,
+                                                uint8_t * pTransactionIdBuffer,
+                                                uint32_t * pSendStunMessageBufferLength );
+
+IceResult_t Ice_CreateRequestForNominatingValidCandidatePair( IceAgent_t * pIceAgent,
+                                                              uint8_t ** ppSendStunMessageBuffer,
+                                                              uint32_t * pSendStunMessageBufferLength,
+                                                              IceCandidatePair_t * pIceCandidatePair,
+                                                              uint8_t * pTransactionIdBuffer );
+
+IceResult_t Ice_CreateRequestForConnectivityCheck( IceAgent_t * pIceAgent,
+                                                   uint8_t ** ppSendStunMessageBuffer,
+                                                   uint32_t * pSendStunMessageBufferLength,
+                                                   uint8_t * pTransactionIdBuffer,
+                                                   IceCandidatePair_t * pIceCandidatePair );
+
+IceResult_t Ice_CreateResponseForRequest( IceAgent_t * pIceAgent,
+                                          uint8_t ** ppSendStunMessageBuffer,
+                                          uint32_t * pSendStunMessageBufferLength,
+                                          IceIPAddress_t * pSrcAddr,
+                                          uint8_t * pTransactionIdBuffer );
+
+IceResult_t Ice_DeserializeStunPacket( StunContext_t * pStunCxt,
```

Comment:
Same here. Can we move this to other header file to avoid user including this unexpectly?

Reply:
updated.